src/go/k8s/controllers/redpanda/console_controller.go

```diff
@@ -91,7 +91,7 @@ func (r *ConsoleReconciler) Reconcile(
 	if err != nil {
 		// Create event instead of logging the error, so user can see in Console CR instead of checking logs in operator
 		switch {
-		case apierrors.IsNotFound(err) || !cluster.DeletionTimestamp.IsZero():
+		case apierrors.IsNotFound(err) || cluster != nil && !cluster.DeletionTimestamp.IsZero():
```

Comment:
Isn't cluster guaranteed to be not nil if GetCluster doesn't return an error?

Reply:
Right, but here we are checking the error is not nil, so there is no guarantee that the cluster object is not nil.

Reply:
> Isn't cluster guaranteed to be not nil if GetCluster doesn't return an error?

Yes, but we are in the place where error was returned.
https://github.com/redpanda-data/redpanda/blob/f513a0534603162bac171ddb346e6d5043cec1b7/src/go/k8s/controllers/redpanda/console_controller.go#L90-L94

In the snippet above ☝️ the line 91.

Then we are in the case where nil will be returned. 
https://github.com/redpanda-data/redpanda/blob/f513a0534603162bac171ddb346e6d5043cec1b7/src/go/k8s/apis/redpanda/v1alpha1/console_types.go#L262
https://github.com/redpanda-data/redpanda/blob/f513a0534603162bac171ddb346e6d5043cec1b7/src/go/k8s/apis/redpanda/v1alpha1/console_types.go#L265